src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -50,42 +106,112 @@ struct DistributedSearchTreeImpl
                             Predicates const &queries,
                             Kokkos::View<int *, DeviceType> &indices,
                             Kokkos::View<int *, DeviceType> &offset,
-                            Kokkos::View<int *, DeviceType> &ranks);
+                            Kokkos::View<int *, DeviceType> &ranks)
+  {
+    Kokkos::View<Kokkos::pair<int, int> *, DeviceType> out("pairs_index_rank",
```

Comment:
Can't remember how this is implemented in Kokkos and if we should still specify we don't want initialization

Reply:
Shouldn't hurt if we are explicit.